central/credentialexpiry/service/service_impl.go

```diff
@@ -77,6 +84,80 @@ func (s *serviceImpl) getCentralCertExpiry() (*v1.GetCertExpiry_Response, error)
 	return &v1.GetCertExpiry_Response{Expiry: expiry}, nil
 }
 
+func (s *serviceImpl) getCentralDBCertExpiry() (*v1.GetCertExpiry_Response, error) {
```

Comment:
This could be simplified a lot by just loading the `central-db-tls` secret and checking the expiration date, but I don't know if Central has permissions to do so (Sensor can do this).

Reply:
Yeah central-db stores the secrets in a table and that has `central-db-tls` too. But I wasn't sure if the secrets will always be named `central-db-tls` for all installation methods. I also tried to see if the cert is volume mounted on central, but it wasn't.

Reply:
Yes, the secret is always called `central-db-tls`, regardless of installation type. However, for external DBs there is no secret (because the TLS certificate is managed by the external DB, not Central).

I think the code below would retrieve the leaf certificate also for external DBs (because it gets it from the `ConnectionState`), but we don't need to refresh that.

Reply:
I changed the implementation to read the cert from `central-db-tls` secret. The previous approach was tied to the protocol specified by postgres for making connection to the db. So the previous approach would have required us to keep up to date with postgres specification for connections.